### ISSUE TITLE
Logins PII Audit

### ISSUE THREAD

**Issue by lougeniaC64** [mozilla/application-services#2407]
This work entails auditing the logins [`import_multiple`](https://github.com/mozilla/application-services/blob/master/components/logins/src/db.rs#L512) function and ensuring that no PII will be captured in the returned metrics JSON object *or* that we've prevented any occurrences of this. Specifically the errors reported [here](https://github.com/mozilla/application-services/blob/master/components/logins/src/db.rs#L586) and [here](https://github.com/mozilla/application-services/blob/master/components/logins/src/db.rs#L623).

┆Issue is synchronized with this [Jira Story](https://jira.mozilla.com/browse/SYNC-544)
┆Story Points: 3
┆Epic: Fenix Migration
┆Sprint: SYNC - end 2020-01-03

**Comment by data-sync-user** [comment]
➤ Ryan Kelly commented:

Re: PII, we want to make sure we're avoiding anything above a "Category 2" as defined here: https://wiki.mozilla.org/Firefox/Data_Collection ( https://wiki.mozilla.org/Firefox/Data_Collection )

(We're generally OK about including Category 3 data in log files, but in this case we're talking about data that goes to telemetry, so we need to be more careful. Category 4 data is right out).

**Comment by rfk** [comment]
I took some time to dig in here. It doesn't look like logins component errors can contain any outright PII, but they can contain information that goes beyond Category 2 for telemetry purposes:

* Error strings can include guids, meaning they can identify a particular login record. If we took the guid of a login record from a telemetry ping and went looking for a corresponding guid in the data stored on the sync server, we might be able to associate that telemetry ping with a specific sync user.
* Error strings for url-parsing failures look like they might include details about the URL that was being parsed, which would be Category 3 data.
* JSON parse errors likewise look like they might include details about the JSON that failed to parse, which could include Category 3 data.

We're generally OK with including such info in local log output, but I don't think we could or should include it in telemetry. Given that the data we're returning here is specifically for inclusion in a future telemetry ping, I'm going to put up a PR that reduces the amount of info logged in these error messages.
